ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp

```diff
@@ -308,7 +308,7 @@ bool window_get_stayontop() {
 
 void window_set_sizeable(bool sizeable) {
   if (window_get_maximized()) return;
-  if (window_get_fullscreen()) return;
+  if (enigma::isFullScreen) return;
```

Comment:
Is this part of the bug fix? This change implies that our internal representation of whether the game is fullscreen _will_ differ from the external representation, _**and**_ that when this happens, we should trust our own understanding over the WM's. Why is that the case? I would expect this situation to arise when the user tells their window manager to adjust the game's fullscreen state instead of telling the game to do so, and in that case, we would want to fix our internal state.

Reply:
It only differs during game window creation. Yes, this is part of the fix. enigma::isFullScreen is based on the global game setting. window_get_fullscreen() is based on state of the fullscreen atom. Since we initially start the game in a window, (and then make it fullscreen relative to the GGS) this is why it happens to work this way.

Reply:
My question is why that's an issue, are you trying to call this in the create event? If so, maybe we need to rethink at what time we apply the GGS. Otherwise we could change the order that fullscreen and sizeable is applied from GGS so that fullscreen is applied first.

Reply:
@RobertBColton my guess Robert as to why this is an issue is due to the fact Ubuntu Budgie has a retarded WM. That's the only explanation I can think of. The issue this pr fixes doesn't happen with most WM's and platforms. Only reproducible in Ubuntu Budgie so far, but my guess is there are many other distro's using that WM.

Reply:
Why don't you just try moving the GGS fullscreen call above the sizeable call please.
https://github.com/enigma-dev/enigma-dev/blob/c7cb43b7db8d3943079d9433e7d37ef7b336a52e/ENIGMAsystem/SHELL/Universal_System/Resources/loading.cpp#L141

Reply:
I already know the technicalities as to why that would undo what i went out of my way to fix completely, but sure, i'll try it .

Reply:
No I mean, test it with this and on master.

Reply:
Pretty much what I was already doing.

Reply:
Well ok, try it then.

Reply:
I am giving you the benefit of a doubt, even though by default unless you tell the window manager you want a fixed sized window, it will be sizeable, which defeats the idea behind what you trying to explain, unless i misread something, which could be the case.

Reply:
Room set to 640 by 480 before and after switching out of fullscreen in master with this code in loading.cpp as i already predicted undid my efforts.

```
    // apply global game settings
    enigma_user::window_set_fullscreen(isFullScreen);
    enigma_user::window_set_sizeable(isSizeable);
    enigma_user::window_set_showborder(showBorder);
```
(dual monitor setup)

Before:
![Screenshot from 2020-05-27 19-05-30](https://user-images.githubusercontent.com/4379204/83080842-4a6ffa80-a04d-11ea-82a0-15dd86476237.png)

After:
![Screenshot from 2020-05-27 19-05-35](https://user-images.githubusercontent.com/4379204/83080853-4e9c1800-a04d-11ea-82d2-4d001ca68201.png)

Reply:
The code we have now is near perfection for all window managers. what's not to like?

Reply:
@RobertBColton I apologize if my posts above come across rude or snippy, that wasn't my intent. I'm trying to be more sensitive to these things.

Reply:
Maybe it would help you to look at the underlying code in SDL which deals with these issues.
https://fossies.org/linux/SDL2/src/video/x11/SDL_x11window.c

Take a look at specifically lines 925-960, 1259-1341, and 1345-1419.

Reply:
And also it would be useful to know if SDL reproduces these issues on Ubuntu Budgie or not.

Reply:
` 1279             if (fullscreen) {
 1280                 /* we are going fullscreen so turn the flags off */
 1281                 sizehints->flags &= ~(PMinSize | PMaxSize);
 1282             } else {
 1283                 /* Reset the min/max width height to make the window non-resizable again */
 1284                 sizehints->flags |= PMinSize | PMaxSize;
 1285                 sizehints->min_width = sizehints->max_width = window->windowed.w;
 1286                 sizehints->min_height = sizehints->max_height = window->windowed.h;
 1287             }`

They appear to be doing the same exact thing I am here, only they are removing the min/max size hints to allow fullscreen rather than making the window sizeable the other way, which is set the min/max to INT_MAX. When they switch out of fullscreen, they restore the window to the size it previously was recorded to be before going fullscreen, which I also do in this pr.

So the educated guess would be, since they are doing the same thing effectively, just a very slightly different means to accomplish the same desired outcome, SDL doesn't have this issue, although I'll go ahead and test it, I seriously doubt they fucked it up like ISM did.

Reply:
@RobertBColton i uninstalled Ubuntu Budgie last night because I got tired of using it, are you going to require i install it again and test with SDL before this will get merged? I will if i have to. I noticed some regressions with mg_keystroke's game, but I'm not sure if this pr causes it at all, I'm going to check out master again and see whether it's a new regression in this pr.